src/include/index/scan_optimizer.h

```diff
@@ -0,0 +1,604 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Peloton
+//
+// index.h
+//
+// Identification: src/include/index/scan_optimizer.h
+//
+// Copyright (c) 2015-16, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <vector>
+
+#include "index/index.h"
+#include "common/exception.h"
+#include "common/logger.h"
+#include "common/pool.h"
+#include "catalog/schema.h"

```

Comment:
I saw that you have some fast forwards below. You could remove some includes if they're not necessary. Or move the definition of complex functions to a .cpp file if they require many dependencies.